src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -1241,7 +1241,14 @@ public static bool IsGraphQLReservedName(Entity entity, string databaseColumnNam
             string schemaName,
             string tableName)
         {
-            DataTable? dataTable = EntitiesDataSet.Tables[tableName];
+            // Because we have an instance of SqlMetadataProvider for each individual database
+            // (note: this means each actual database not each database type), we do not
+            // need to worry about collisions beyond that schema, hence no database name is needed.
+            string tableNameWithPrefix = GetTableNameWithPrefix(
+                databaseName: string.Empty,
+                schemaName: schemaName,
+                tableName: tableName);
+            DataTable? dataTable = EntitiesDataSet.Tables[tableNameWithPrefix];
```

Comment:
if the key is not found, I think this will throw exception. Instead of directly indexing the Dictionary in this manner, I think we should use `TryGet`

Reply:
DataTable doesn't have a `TryGet`, and if the key is not found it simply returns an empty data table.